sdv/evaluation/single_table.py

```diff
@@ -50,7 +58,14 @@ def run_diagnostic(real_data, synthetic_data, metadata, verbose=True):
             Single table diagnostic report object.
     """
     diagnostic_report = DiagnosticReport()
-    diagnostic_report.generate(real_data, synthetic_data, metadata.to_dict(), verbose)
+    if not isinstance(metadata, Metadata):
+        metadata = Metadata()._convert_to_unified_metadata(metadata)
+
+    if len(metadata.tables) > 1:
+        raise ValueError('Only a single table is allowed in metadata.')
+
+    metadata_dict = next(iter(metadata.tables.values())).to_dict()
```

Comment:
In the case where it is the old metadata, we convert it to the new one by converting to a cit, and then loading from that. Then we go back to a dict here. Maybe we should just convert to a dict and raise the warning instead of converting back and forth